### PR TITLE
Sprint2 ko unit test

### PR DESCRIPTION
**Pull request by kkaungyyang**

### CODE REVIEW

File: web-component/inputBox.js [hunk 1]
```diff
@@ -29,6 +29,53 @@ class InputBox extends HTMLElement {
         }
```
Comment (ericsengithub) [inline web-component/inputBox.js]: Not reviewing this portion to be deleted in another pull request, ideally all the functionality wiill be placed into inputRT

File: web-component/inputRT.js [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
     };
 
+    const SIZES = {
+        "l": "width: 500px; height: 50px; font-size: 25px !important; padding: 10px 10px;",
+        "m": "width: 400px; height: 40px; font-size: 20px !important; padding: 8px 8px;",
+        "d": "width: 300px; height: 30px; font-size: 18px !important;",
+        "s": "width: 250px; height: 30px; font-size: 15px !important; padding: 5px 5px;"
+    };
+
+    let currentMode = "input";
```
Comment (ericsengithub) [inline web-component/inputRT.js]: please delete this

File: web-component/inputRT.js [hunk 3]
```diff
@@ -63,6 +74,230 @@
...
+            if (isPassword) {
+                this.setAttribute('password', val);
+            } else {
+                this.removeAttribute('password');
+            }
+        }
+
+        _init_password(){
+            if (this.password) {
+                if(this.mode !== null){
```
Comment (ericsengithub) [inline web-component/inputRT.js]: please change this to use our personally defined function

Comment (ericsengithub) [reply web-component/inputRT.js]: _choose_element(mode)

File: web-component/inputRT.js [hunk 4]
```diff
@@ -63,6 +74,230 @@
...
+            return this.getAttribute('width');
+        }
+        /**
+         * set width(val) 
+         * Sets width if value passed in, or removes it if nothing
+         * is passed.
+         * Returns: Null
+         */
+        set width(val) {
+            const isWidth = String(val);
```
Comment (ericsengithub) [inline web-component/inputRT.js]: We should discuss of this type of casting is the right way to do it

File: web-component/inputRT.js [hunk 5]
```diff
@@ -63,6 +74,230 @@
...
+                this.removeAttribute('width');
+            }
+        }
+
+        _init_width(){
+            if (this.width) {
+                const sizeStyle = `input {
+                    width: ${this.width} !important; 
+                }`;
+                this.shadowRoot.querySelector('style').innerHTML += sizeStyle;
```
Comment (ericsengithub) [inline web-component/inputRT.js]: I know we tried it with Olivia but lets try to brainstorm a way to use append rather than modifying the raw innerHTML

Comment (ericsengithub) [reply web-component/inputRT.js]: or appendChild?

File: web-component/inputRT.js [hunk 6]
```diff
@@ -63,6 +74,230 @@
...
+            }
+        }
+
+        _init_size() {
+            if (this.size && ! this.width && ! this.height) {
+                const sizeStyle = `input {
+                    ${SIZES[this.size]}
+                }`;
+                this.shadowRoot.querySelector('style').innerHTML += sizeStyle;
+            } else if (!this.size && !this.width && !this.height) { // default size
```
Comment (ericsengithub) [inline web-component/inputRT.js]: I don't think we need this else if, just let the code flow through

File: web-component/inputRT.js [hunk 7]
```diff
@@ -63,6 +74,230 @@
...
+
+        _init_disabled(){
+            if(this.disabled){
+                const disabledStyle = `input {
+                    opacity: 0.5!important;
+                    cursor: not-allowed;
+                    background-color: #ccc;
+                }`; 
+                //input.setAttribute('disabled', true) TO DO: is this necessary? already true to begin with
+                this.shadowRoot.querySelector('style').innerHTML += disabledStyle;
```
Comment (ericsengithub) [inline web-component/inputRT.js]: same idea with the comment mentioned in init_width

File: web-component/inputRT.js [hunk 8]
```diff
@@ -97,6 +332,27 @@
...
 
+        /**
+         * Internal function to determine the correct element in concern. 
+         * @param {*} mode 
+         * @returns {string} an html element in concern
+         */
+        _choose_element(mode) {
+            let retVal = "";
+            switch (mode) {
+                case "custom":
```
Comment (ericsengithub) [inline web-component/inputRT.js]: Perhaps a comment with todo about brainstorming a design discussion on how to handle custom

File: web-component/inputRT.js [hunk 9]
```diff
@@ -154,11 +151,13 @@
...
+        /**
+         * Initializes this element to be of type password only if the element in concern is input
+         */
         _init_password(){
             if (this.password) {
-                if(this.mode !== null){
-                    this._textSlot.querySelector(this.mode).setAttribute('type', "password");
-                } else {
+                const el = this._choose_element(this.mode);  
+                if(el && el === "input") {
```
[the remark this answers is not in the document]
Comment (okwong) [reply web-component/inputRT.js]: need to make sure el exists and also that it = "input"
